src/User.php

```diff
@@ -4990,6 +5036,10 @@ public static function getOrImportByEmail($email = '')
                     }
                 }
             }
+            if ($createuserfromemail) {
```

Comment:
IMHO, this logic may produce unexpected result when `is_users_auto_add` is set to false. Indeed, user will not be imported from LDAP, but will be created anyway, and maybe with a login that differs from the LDAP value. I do not really know how it will be handled after that in LDAP sync process, but I guess it could result in having 2 users with the same email, the first created from mail collector and the second created from LDAP sync.